Sources/MongoSwift/Operations/WatchOperation.swift

```diff
@@ -24,7 +24,7 @@ internal struct WatchOperation<CollectionType: Codable, ChangeStreamType: Codabl
                   options: ChangeStreamOptions?,
                   stealing connection: Connection) throws {
         self.target = target
-        self.pipeline = pipeline
+        self.pipeline = .array(pipeline.map { .document($0) })
```

Comment:
I think what you said generally holds. If we need to use a value for its `BSONValue`, don't put it in a `BSON` unless we need to (e.g. for putting into a document).

In this case, we don't need the pipeline except to put it into a document, so I figured it could just be mapped to `BSON` in the constructor. If we needed to inspect the pipeline in `execute`, we would want to keep it as `[Document]`.